### PR TITLE
Create stringtemplate4_ubuntu_16.04.sh

### PR DESCRIPTION
**Pull request by nageshlop**
Created shell script to run the build and verified the working on local VM.

### CODE REVIEW

**Comment by nageshlop** [comment]
Build logs:

ubuntu@nagesh-at:/scratch/jar-proj$ ./string-template4.sh master
Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
Ign:2 http://ppa.launchpad.net/beineri/opt-qt591-trusty/ubuntu focal InRelease
Get:3 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease [114 kB]
............
.......
...
Buildfile: /scratch/jar-proj/stringtemplate4/build.xml

basic-init:

clean:

build-init:

antlr3-init:
      [get] Destination already exists (skipping): /scratch/jar-proj/stringtemplate4/lib/antlr-3.5.2-complete.jar

antlr3-up-to-date:

antlr3:
    [mkdir] Created dir: /scratch/jar-proj/stringtemplate4/build/generated-sources/antlr3
     [echo] Generating ANTLR 3 grammars:
     [echo]   org/stringtemplate/v4/compiler/CodeGenerator.g
     [echo]   org/stringtemplate/v4/compiler/Group.g
     [echo]   org/stringtemplate/v4/compiler/STParser.g
    [mkdir] Created dir: /scratch/jar-proj/stringtemplate4/build/generated-sources/antlr3/org/stringtemplate/v4/compiler
    [touch] Creating /scratch/jar-proj/stringtemplate4/build/antlr3-3.5.2.touch

up-to-date:

compile:
    [mkdir] Created dir: /scratch/jar-proj/stringtemplate4/build/classes
    [javac] Compiling 63 source files to /scratch/jar-proj/stringtemplate4/build/classes
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 6
    [javac] warning: [options] source value 6 is obsolete and will be removed in a future release
    [javac] warning: [options] target value 1.6 is obsolete and will be removed in a future release
    [javac] warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/STGroup.java:750: warning: [unchecked] unchecked cast
    [javac]         return (ModelAdaptor<? super T>) adaptors.get(attributeType);
    [javac]                                                      ^
    [javac]   required: ModelAdaptor<? super T>
    [javac]   found:    ModelAdaptor<CAP#1>
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in method <T>getModelAdaptor(Class<T>)
    [javac]   where CAP#1 is a fresh type-variable:
    [javac]     CAP#1 extends Object from capture of ?
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/STGroup.java:799: warning: [unchecked] unchecked cast
    [javac]         return (AttributeRenderer<? super T>) renderers.get(attributeType);
    [javac]                                                            ^
    [javac]   required: AttributeRenderer<? super T>
    [javac]   found:    AttributeRenderer<CAP#1>
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in method <T>getAttributeRenderer(Class<T>)
    [javac]   where CAP#1 is a fresh type-variable:
    [javac]     CAP#1 extends Object from capture of ?
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:220: warning: [rawtypes] found raw type: DefaultListModel
    [javac]             final DefaultListModel errorListModel = new DefaultListModel();
    [javac]                   ^
    [javac]   missing type arguments for generic class DefaultListModel<E>
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class DefaultListModel
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:220: warning: [rawtypes] found raw type: DefaultListModel
    [javac]             final DefaultListModel errorListModel = new DefaultListModel();
    [javac]                                                         ^
    [javac]   missing type arguments for generic class DefaultListModel<E>
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class DefaultListModel
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:222: warning: [unchecked] unchecked call to addElement(E) as a member of the raw type DefaultListModel
    [javac]                 errorListModel.addElement(msg);
    [javac]                                          ^
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class DefaultListModel
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:224: warning: [unchecked] unchecked call to setModel(ListModel<E>) as a member of the raw type JList
    [javac]             viewFrame.errorList.setModel(errorListModel);
    [javac]                                         ^
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class JList
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:244: warning: [rawtypes] found raw type: ListModel
    [javac]                         ListModel model = viewFrame.errorList.getModel();
    [javac]                         ^
    [javac]   missing type arguments for generic class ListModel<E>
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in interface ListModel
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViz.java:421: warning: [deprecation] modelToView(int) in JTextComponent has been deprecated
    [javac]                     comp.scrollRectToVisible(comp.modelToView(i));
    [javac]                                                  ^
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/Interpreter.java:1221: warning: [rawtypes] found raw type: ModelAdaptor
    [javac]             ModelAdaptor adap = self.groupThatCreatedThisInstance.getModelAdaptor(o.getClass());
    [javac]             ^
    [javac]   missing type arguments for generic class ModelAdaptor<T>
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in interface ModelAdaptor
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/Interpreter.java:1222: warning: [unchecked] unchecked call to getProperty(Interpreter,ST,T,Object,String) as a member of the raw type ModelAdaptor
    [javac]             return adap.getProperty(this, self, o, property, toString(out,scope,property));
    [javac]                                    ^
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in interface ModelAdaptor
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViewFrame.java:69: warning: [rawtypes] found raw type: JList
    [javac]         errorList = new JList();
    [javac]                         ^
    [javac]   missing type arguments for generic class JList<E>
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class JList
    [javac] /scratch/jar-proj/stringtemplate4/src/org/stringtemplate/v4/gui/STViewFrame.java:195: warning: [rawtypes] found raw type: JList
    [javac]     protected JList errorList;
    [javac]               ^
    [javac]   missing type arguments for generic class JList<E>
    [javac]   where E is a type-variable:
    [javac]     E extends Object declared in class JList
    [javac] 16 warnings

zip-source:
    [mkdir] Created dir: /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9
    [mkdir] Created dir: /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9/src
     [copy] Copying 70 files to /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9/src
     [copy] Copying 3 files to /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9
     [copy] Copying 1 file to /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9/lib
      [zip] Building zip: /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9-src.zip

build-jar:
      [jar] Building jar: /scratch/jar-proj/stringtemplate4/dist/ST-4.0.9.jar

distribute:

BUILD SUCCESSFUL

basic-init:

clean:
   [delete] Deleting directory /scratch/jar-proj/stringtemplate4/build
   [delete] Deleting directory /scratch/jar-proj/stringtemplate4/dist

BUILD SUCCESSFUL

File: s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
+https://github.com/ppc64le/build-scripts/pull/875
```
Comment (gururajrkatti) [inline s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh:1]: what is the extra line here ?

File: s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+PKG_NAME=${URL##*/}
+
+if [ -d $PKG_NAME ] ; then
+  rm -rf $PKG_NAME
+fi
+
+#Clean and Install Updates
+sudo apt-get clean
+sudo apt-get update
+sudo apt-get install maven
```
Comment (gururajrkatti) [inline s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh:33]: Is it required since we are not doing mvn build ? Also isn't ant need to be installed ?

**Review by gururajrkatti — COMMENTED** [review]
Don't we need to run any tests here ?

File: s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh (hunk 1 above)
Comment (nageshlop) [reply s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh]: Have removed its typo. Didn't find any tests.

File: s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh (hunk 2 above)
Comment (nageshlop) [reply s/stringtemplate4/stringtemplate4_ubuntu_16.04.sh]: Updated the script.

**Comment by nageshlop** [comment]
> Don't we need to run any tests here ?

@gururajrkatti, target test doesn't exists.

**Review by gururajrkatti — APPROVED** [review]
LGTM

**Pull request merged**